core/src/main/scala/caliban/execution/Executor.scala

```diff
@@ -57,9 +58,9 @@ object Executor {
           value match {
             case EnumValue(v) =>
               // special case of an hybrid union containing case objects, those should return an object instead of a string
-              val obj = mergeFields(currentField, v).collectFirst {
+              val obj = currentField.fields.collectFirst {
```

Comment:
@ghostdogpr do you see a reason why we'd need to merge the fields of an enum value here? Tests seem to pass without it and I'm not sure I can think of a case where we'd need to merge them.

If you can think of a case where this is needed let me know and I'll rollback the change and add a test instead

Reply:
Yeah I don't think it's needed considering the `collectFirst` right after.

Reply:
Yeah that's what I thought as well. The only thing that might be needed is the filtering we do within `mergeFields`. Do you think that might be needed here?

```scala
if (field._condition.forall(_.contains(typeName))) { ... }
```

Reply:
Ah... maybe if there is a fragment that applies only to a subtype of the enum?

Reply:
Just to be safe, I added back the filtering using a `.view` without  calling `mergeFields`. This way we avoiding filtering the entire collection in cases that it's not needed. I think the performance impact should be extremely minimal either way

Reply:
Sounds good! And feel free to merge when CI is green 👍